cranelift/codegen/src/isa/x64/abi.rs

```diff
@@ -1,73 +1,584 @@
+    fn get_caller_saves(call_conv: isa::CallConv) -> Vec<Writable<Reg>> {
+        let mut caller_saved = Vec::new();
 
-    /// Clobbered registers, as indicated by regalloc.
-    clobbered: Set<Writable<RealReg>>,
+        // Systemv calling convention:
+        // - GPR: all except RBX, RBP, R12 to R15 (which are callee-saved).
+        caller_saved.push(Writable::from_reg(regs::rsi()));
+        caller_saved.push(Writable::from_reg(regs::rdi()));
+        caller_saved.push(Writable::from_reg(regs::rax()));
+        caller_saved.push(Writable::from_reg(regs::rcx()));
+        caller_saved.push(Writable::from_reg(regs::rdx()));
+        caller_saved.push(Writable::from_reg(regs::r8()));
+        caller_saved.push(Writable::from_reg(regs::r9()));
+        caller_saved.push(Writable::from_reg(regs::r10()));
+        caller_saved.push(Writable::from_reg(regs::r11()));
 
-    /// Total number of spill slots, as indicated by regalloc.
-    num_spill_slots: Option<usize>,
+        if call_conv.extends_baldrdash() {
+            caller_saved.push(Writable::from_reg(regs::r12()));
+            caller_saved.push(Writable::from_reg(regs::r13()));
+            // Not r14; implicitly preserved in the entry.
+            caller_saved.push(Writable::from_reg(regs::r15()));
+            caller_saved.push(Writable::from_reg(regs::rbx()));
+        }
 
-    /// Calculated while creating the prologue, and used when creating the epilogue. Amount by
-    /// which RSP is adjusted downwards to allocate the spill area.
-    frame_size_bytes: Option<usize>,
+        // - XMM: all the registers!
+        caller_saved.push(Writable::from_reg(regs::xmm0()));
+        caller_saved.push(Writable::from_reg(regs::xmm1()));
+        caller_saved.push(Writable::from_reg(regs::xmm2()));
+        caller_saved.push(Writable::from_reg(regs::xmm3()));
+        caller_saved.push(Writable::from_reg(regs::xmm4()));
+        caller_saved.push(Writable::from_reg(regs::xmm5()));
+        caller_saved.push(Writable::from_reg(regs::xmm6()));
+        caller_saved.push(Writable::from_reg(regs::xmm7()));
+        caller_saved.push(Writable::from_reg(regs::xmm8()));
+        caller_saved.push(Writable::from_reg(regs::xmm9()));
+        caller_saved.push(Writable::from_reg(regs::xmm10()));
+        caller_saved.push(Writable::from_reg(regs::xmm11()));
+        caller_saved.push(Writable::from_reg(regs::xmm12()));
+        caller_saved.push(Writable::from_reg(regs::xmm13()));
+        caller_saved.push(Writable::from_reg(regs::xmm14()));
+        caller_saved.push(Writable::from_reg(regs::xmm15()));
 
-    call_conv: CallConv,
+        caller_saved
+    }
+}
 
-    /// The settings controlling this function's compilation.
-    flags: settings::Flags,
+impl From<StackAMode> for SyntheticAmode {
+    fn from(amode: StackAMode) -> Self {
+        match amode {
+            StackAMode::FPOffset(off, _ty) => {
+                let off = i32::try_from(off).unwrap();
```

Comment:
nit: could you use `expect` instead of `unwrap`, here and below? it likely adds cases where we have a very big offset, and it just wouldn't be implemented here.

Reply:
Done!